packages/site-admin/.storybook/mock-registry/index.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import { createReduxStore, createRegistry } from '@wordpress/data';
+
+/**
+ * Registers a mock store with the given initial state.
+ */
+const registerMockStore = ( registry, storeName, initialState ) => {
+	const store = createReduxStore( storeName, {
+		reducer: ( state = initialState ) => state,
+		selectors: {
+			getEntityRecord: ( state, kind, name ) => {
+				if ( kind === 'root' && name === '__unstableBase' ) {
+					return state[ 'root/__unstableBase' ] || null;
+				}
+				return null;
+			},
+		},
+		actions: {},
+	} );
+
+	registry.register( store );
+};
+
+/**
+ * Creates a global mock registry for @wordpress/data
+ */
+export const createMockRegistry = ( stores = {} ) => {
```

Comment:
Is there a good reason to have a default `stores` value? I don't think so. Some improved types could help ensure we pass a `stores`.

```suggestion
export const createMockRegistry = ( stores ) => {
```

Reply:
Going to address it in a follow-up.

Reply:
#99710